src/Components/ServiceScene/ServiceScene.tsx

```diff
@@ -69,41 +78,76 @@ export function getDetectedLabelsFrame(sceneRef: SceneObject) {
   return serviceScene.state.$detectedLabelsData?.state.data?.series?.[0];
 }
 
+export function getDetectedFieldsFrame(sceneRef: SceneObject) {
+  const serviceScene = sceneGraph.getAncestor(sceneRef, ServiceScene);
+  return getDetectedFieldsFrameFromQueryRunnerState(serviceScene.state.$detectedFieldsData?.state);
+}
+
+export const getDetectedFieldsFrameFromQueryRunnerState = (state?: QueryRunnerState) => {
+  // Only ever one frame in the response
+  return state?.data?.series?.[0];
+};
+
+export const getDetectedFieldsNamesFromQueryRunnerState = (state: QueryRunnerState) => {
+  // The first field, DETECTED_FIELDS_NAME_FIELD, has the list of names of the detected fields
+  return state.data?.series?.[0]?.fields?.[0];
+};
+
 export class ServiceScene extends SceneObjectBase<ServiceSceneState> {
   protected _variableDependency = new VariableDependencyConfig(this, {
     variableNames: [VAR_DATASOURCE, VAR_LABELS, VAR_FIELDS, VAR_PATTERNS, VAR_LEVELS],
   });
 
   public constructor(
-    state: MakeOptional<ServiceSceneState, 'body' | '$data' | '$patternsData' | '$detectedLabelsData'>
+    state: MakeOptional<
+      ServiceSceneState,
+      'body' | '$data' | '$patternsData' | '$detectedLabelsData' | '$detectedFieldsData' | 'loadingStates'
+    >
   ) {
     super({
+      loadingStates: {
+        [TabNames.patterns]: false,
+        [TabNames.labels]: false,
+        [TabNames.fields]: false,
+        [TabNames.logs]: false,
+      },
       loading: true,
       body: state.body ?? buildGraphScene(),
       $data: getServiceSceneQueryRunner(),
       $patternsData: getPatternsQueryRunner(),
       $detectedLabelsData: getDetectedLabelsQueryRunner(),
+      $detectedFieldsData: getDetectedFieldsQueryRunner(),
       ...state,
     });
 
     this.addActivationHandler(this.onActivate.bind(this));
   }
 
-  private setEmptyFiltersRedirection() {
+  private setSubscribeToLabelsVariable() {
     const variable = getLabelsVariable(this);
     if (variable.state.filters.length === 0) {
       this.redirectToStart();
       return;
     }
     this._subs.add(
-      variable.subscribeToState((newState) => {
+      variable.subscribeToState((newState, prevState) => {
+        const newServiceName = getServiceNameFromVariableState(newState);
+        const prevServiceName = getServiceNameFromVariableState(prevState);
         if (newState.filters.length === 0) {
           this.redirectToStart();
         }
         // If we remove the service name filter, we should redirect to the start
         if (!newState.filters.some((f) => f.key === SERVICE_NAME)) {
           this.redirectToStart();
         }
+
+        // Clear filters if changing service, they might not exist, or might have a different parser
```

Comment:
This is a slight degradation in UX, ideally we would check the detected_fields response against the new service and update parsers. Should create an issue for this